src/render/painter.js

```diff
@@ -748,17 +749,17 @@ class Painter {
         ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, layerTimer.query);
     }
 
-    gpuTimingGlobeStart() {
-        if (this.options.gpuTimingGlobe) {
+    gpuTimingDeferredRenderStart() {
+        if (this.options.gpuTimingDeferredRender) {
             const ext = this.context.extTimerQuery;
-            if (!this.globeGpuTimeQuery)
-                this.globeGpuTimeQuery = ext.createQueryEXT();
-            ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, this.globeGpuTimeQuery);
+            const query = ext.createQueryEXT();
+            this.deferredRenderGpuTimeQueries.push(query);
```

Comment:
This is really interesting! So basically we add as many queries as necessary and ensure that by the end of the measurement we have at least one that correctly captures the elapsed time. In practice this loop shouldn't be executed too many times right?

Reply:
Yes, as far as I understand it's possible to add as many queries as you'd want within the same frame and collect once. The only thing that I fixed in this PR is fixing the reset, as we have the 50ms timer. I wonder if the better way would be to use QUERY_RESULT_AVAILABLE_EXT instead of using a timer, but it seems to work either way.

Reply:
> So basically we add as many queries as necessary and ensure that by the end of the measurement we have at least one that correctly captures the elapsed time.

I believe the idea is to measure total time spent in `drawTerrainRaster` rather than just measuring a single call.

Reply:
Yes indeed. I meant to say It would have been ideal if a query would accumulate the time readings in which case we wouldn't need to add many of them. But this as seen here isn't a big deal :)

Reply:
> I wonder if the better way would be to use QUERY_RESULT_AVAILABLE_EXT instead of using a timer, but it seems to work either way.

@karimnaaji we could if we make use of two sets of queries but I guess in practice what we have here is really sufficient.